package/MDAnalysis/transformations/__init__.py

```diff
@@ -26,34 +26,77 @@
 Trajectory transformations --- :mod:`MDAnalysis.transformations`
 ================================================================
 
-The transformations submodule contains a collection of functions to modify the
-trajectory. Coordinate transformations, such as PBC corrections and molecule fitting
-are often required for some analyses and visualization, and the functions in this
-module allow transformations to be applied on-the-fly.
-These transformation functions can be called by the user for any given
-timestep of the trajectory, added as a workflow using :meth:`add_transformations`
-of the :mod:`~MDAnalysis.coordinates.base` module, or upon Universe creation using
+The transformations submodule contains a collection of function-like classes to
+modify the trajectory.
+Coordinate transformations, such as PBC corrections and molecule fitting
+are often required for some analyses and visualization, and the functions in
+this module allow transformations to be applied on-the-fly.
+
+A typical transformation class looks like this:
+
+.. code-blocks:: python
+
+    class transformation(object):
+        def __init__(self, *args, **kwargs):
+            #  do some things
+            #  save needed args as attributes.
+            self.needed_var = args[0]
+
+        def __call__(self, ts):
+            #  apply changes to the Timestep,
+            #  or modify an AtomGroup and return Timestep
+
+            return ts
+
```

Comment:
You showed the abstract class. Now show a concrete example to address @richardjgowers comment. For instance
```suggestion
As a concrete example we will write a transformation that rotates a group of atoms around 
the z-axis through the center of geometry by a fixed increment for every time step. We will use
:meth:`MDAnalysis.core.groups.AtomGroup.rotateby` and simply increment the rotation angle
every time the transformation is called::

  class spin_atoms(object):
      def __init__(self, atoms, dphi):
         """Rotate atoms by dphi degrees for every time step (around the z axis)"""
         self.atoms = atoms
         self.dphi = dphi
         self.axis = np.array([0, 0, 1])
         self.phi = 0
         
     def __call__(self, ts):
         self.atoms.rotateby(self.phi, self.axis)
         return ts

This transformation can be used as ::
  
   u = mda.Universe(PSF, DCD)
   u.trajectory.add_transformations(spin_atoms(u.select_atoms("protein"), 1.0))

```

I currently can't get nglview to work in my notebook so you'll need to check that this actually works... or come up with another example.

Reply:
Thanks! it works. And I think it makes sense to just rotate by a fixed angle. (for visualization perhaps)

Reply:
I changed the snippet.